File: negocio-tundama-backend/src/main/java/com/tundama/negociotundamabackend/model/entity/Cliente.java
package com.tundama.negociotundamabackend.model.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.time.LocalDateTime;

@Entity
@Table(name = "clientes")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Cliente {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "id_cliente", unique = true, nullable = false, length = 50)
    private String idCliente;  // Formato: NTC-2023-ABR-0052

    @Column(name = "nombre", nullable = false, length = 100)
    private String nombre;

    @Column(name = "tipo_identificacion", length = 20)
    private String tipoIdentificacion;  // NIT o CEDULA

    @Column(name = "identificacion", unique = true, length = 50)
    private String identificacion;

    @Column(name = "direccion", length = 200)
    private String direccion;

    @Column(name = "telefono", length = 20)
    private String telefono;

    @Column(name = "notas", length = 500)
    private String notas;

    @Column(name = "fecha_registro")
    private LocalDateTime fechaRegistro;

    @Column(name = "activo")
    private Boolean activo = true;

    @PrePersist
    public void prePersist() {
        fechaRegistro = LocalDateTime.now();
    }
}

File: Version Android/lib/models/cliente.dart
// ==================== lib/models/cliente.dart ====================
class Cliente {
  final int? id;
  final String idCliente;
  final String nombre;
  final String? tipoIdentificacion;
  final String? identificacion;
  final String? direccion;
  final String? telefono;
  final String? notas;
  final DateTime? fechaRegistro;
  final bool activo;

  Cliente({
    this.id,
    required this.idCliente,
    required this.nombre,
    this.tipoIdentificacion,
    this.identificacion,
    this.direccion,
    this.telefono,
    this.notas,
    this.fechaRegistro,
    this.activo = true,
  });

  factory Cliente.fromJson(Map<String, dynamic> json) {
    return Cliente(
      id: json['id'],
      idCliente: json['idCliente'] ?? '',
      nombre: json['nombre'] ?? '',
      tipoIdentificacion: json['tipoIdentificacion'],
      identificacion: json['identificacion'],
      direccion: json['direccion'],
      telefono: json['telefono'],
      notas: json['notas'],
      fechaRegistro: json['fechaRegistro'] != null 
          ? DateTime.parse(json['fechaRegistro']) 
          : null,
      activo: json['activo'] ?? true,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'nombre': nombre,
      'tipoIdentificacion': tipoIdentificacion,
      'identificacion': identificacion,
      'direccion': direccion,
      'telefono': telefono,
      'notas': notas,
    };
  }
}
